docs/tests-ui.rst

```diff
@@ -1,177 +1,148 @@
-.. _Intern: https://theintern.github.io/
+The tests directory comprises of:
 
-Installing Dependencies
------------------------
+   * ``/functional`` contains pytest tests.
+   * ``/pages`` contains Python pypom page objects.
+   * ``/utils`` contains helper functions.
 
-1. Go to the ``tests/ui/`` directory::
+.. _`Selenium`: http://docs.seleniumhq.org/
+.. _`pytest`: http://pytest.org/latest/
 
-    cd tests/ui
+Setting up test virtual environment
+===================================
 
-2. Use ``npm`` to install Intern::
+#. In the terminal go to the directory where you have downloaded kuma
 
-    npm install intern@^3.0
+#. Create a virtual environment for the tests to run in (the example uses the
+   name mdntests)::
 
-.. warning:: Do *not* install Intern globally -- path issues may occur.
+   $ virtualenv mdntests
 
-Running Tests
--------------
+#. Activate the virtual environment::
 
-On your machine
-~~~~~~~~~~~~~~~
+   $ source mdntests/bin/activate
 
-1. Install `JDK <http://www.oracle.com/technetwork/java/javase/downloads/index.html>`_
+#. Install the requirements the tests need to run::
 
-2. Download the most current `release of Selenium <http://selenium-release.storage.googleapis.com/index.html>`_ standalone server. (It's the ``.jar`` file.)
+   $ pip install -r requirements/test.txt
 
-.. note:: Firefox should work out of the box. You need to install `Chrome <https://sites.google.com/a/chromium.org/chromedriver/>`_ and `Safari <https://code.google.com/p/selenium/wiki/SafariDriver>`_ drivers yourself.
+You may want to add your virtual environment folder to your local .gitignore
+file
 
-3. From the command line, start WebDriver::
+Running the tests
+=================
 
-    # Substitute your WebDriver version in the `#` chars
-    java -jar /path/to/selenium-server-standalone-#.#.#.jar
+The minimum amount of information necessary to run the tests against the staging
+server is the directory the tests are in and which driver to use.
 
-4. Go to the ``tests/ui/`` directory::
+In the virtual environment run::
 
-    cd tests/ui
+   $ py.test tests/functional/ --driver Firefox --driver-path /path/to/geckodriver
 
-5. Run intern with the ``intern-local`` config file (omit the ``.js``)::
+This basic command can be modified to run subsets of the tests, use different
+browsers, run the tests against a local environment, or run tests concurrently.
 
-    ./node_modules/.bin/intern-runner config=intern-local b=firefox
+Only running tests in one file
+------------------------------
 
-The above tries to run the entire suite of tests. You can change the behavior with `command line arguments`_. E.g., ::
+Add the name of the file to the test location::
 
-    node_modules/.bin/intern-runner config=intern-local b=firefox,chrome t=auth,homepage d=developer-local.allizom.org u=someone@somewhere.com p=8675309 wd='Web' destructive=true
+   $ py.test tests/functional/test_search.py --driver Firefox --driver-path
+   /path/to/geckodriver
```

Comment:
When running this command, I'm asked if I want ``python2.7`` to accept connections (I accept), and then I'm presented with a blank Firefox window.  After a few seconds, the test fails, with the final error:

```
WebDriverException: Message: Can't load the profile. Profile Dir: /var/folders/61/s6_xxhqd3nl27_vgq9fzjmkr0000gq/T/tmp2ANHVe If you specified a log_file in the FirefoxBinary constructor, check it for details.
```

This is with Firefox 49.0.2.  Full traceback: https://pastebin.mozilla.org/8924594